winch/codegen/src/isa/x64/asm.rs

```diff
@@ -845,6 +845,20 @@ impl Assembler {
         })
     }
 
+    pub fn sqrt(&mut self, src: Reg, dst: Reg, size: OperandSize) {
+        let op = match size {
+            OperandSize::S32 => SseOpcode::Sqrtss,
+            OperandSize::S64 => SseOpcode::Sqrtsd,
+            OperandSize::S128 => unreachable!(),
+        };
+
+        self.emit(Inst::XmmUnaryRmR {
+            op,
+            src: XmmMemAligned::from(Xmm::from(src)),
```

Comment:
There's 

```rust 
impl From<Reg>for Xmm {
    fn from(reg: Reg) -> Self {
        Xmm::new(reg.into()).expect("valid Xmm register")
    }
}
```

so you could use `src.into()` if you wanted to avoid the boilerplate.

Reply:
I wasn't able to reduce it to a single `.into()` call without a compiler error. I can do `Xmm::from(src).into()`. I also tried `XmmMemAlligned::from(src.into())` and that also fails without specifying the types.